### DISCUSSION TITLE
S3VirtualFiles to support HTTP Range Requests

### DISCUSSION THREAD

**Discussion by tiptopweb** [ServiceStack/Discuss#11] [Ideas]
Following this
https://forums.servicestack.net/t/access-to-videos-over-5m/10409

When using S3VirtualFiles (AWS S3 Buckets), we cannot access video files above 5M
There is error with the browser requesting a HTTP Range request, but the ResponseStream returned by S3’s GetObject() API isn’t seekable

It looks like there’s a way to support HTTP Range Requests using S3 SDK ByteRange request but it will require breaking out of the Stream abstraction and implementing new APIs on all VFS providers which will take a while.

**Comment by mythz** [comment]
I've extended `IVirtualFiles` with `WritePartialToAsync` and added custom support for S3 in [this commit](https://github.com/ServiceStack/ServiceStack/commit/1c87274186ecc25f35a7808a3b4696998f74f659) so you should now be able to return partial HTTP Range requests from a S3 Virtual FIle.

This change is now available from v6.0.3 that's now [available on MyGet](https://docs.servicestack.net/myget).
